src/components/calendar/index.js

```diff
@@ -358,41 +372,52 @@ module.exports = {
         }
         return "default";
     },
-    updateCalendarEvent: function(calendar, calendarName, year, month, id, calendarEvent) {
-        let dirPath = this.findCalendarDirectory(calendarName) + "/" + year + "/" + month;
-        let filename = id + this.CALENDAR_FILE_EXTENSION;
+    updateCalendarEvent: function(calendar, item) {
+        let calendarDirectory = this.findCalendarDirectory(item.calendarName);
+        let dirPath =  item.isRecurring ? calendarDirectory + "/recurring" : calendarDirectory + "/" + item.year + "/" + item.month;
+        let filename = item.Id + this.CALENDAR_FILE_EXTENSION;
         let filePath = peergos.client.PathUtils.toPath(dirPath.split('/'), filename);
         let encoder = new TextEncoder();
-        let uint8Array = encoder.encode(calendarEvent);
+        let uint8Array = encoder.encode(item.item);
         let bytes = convertToByteArray(uint8Array);
         return calendar.writeInternal(filePath, bytes);
     },
     saveEvent: function(calendar, item) {
 	    const that = this;
 	    that.displaySpinner();
-	    if(item.calendarName == item.previousCalendarName) {
-            this.updateCalendarEvent(calendar, item.calendarName, item.year, item.month, item.Id, item.item).thenApply(function(res) {
-                that.removeSpinner();
-            }).exceptionally(function(throwable) {
-                that.showMessage("Unable to save event","Please close calendar and try again");
-                console.log(throwable.getMessage());
-                that.removeSpinner();
-            });
-        } else { //move between calendars
-            this.removeCalendarEvent(calendar, item.previousCalendarName, item.year, item.month, item.Id).thenApply(function(res) {
-                that.updateCalendarEvent(calendar, item.calendarName, item.year, item.month, item.Id, item.item).thenApply(function(res2) {
+	    if (item.action == "createRecurring") {
+            this.moveEvent(calendar, item, false);
+	    } else if (item.action == "deleteRecurring") {
+            this.moveEvent(calendar, item, true);
+	    } else {
+            if(item.calendarName == item.previousCalendarName) {
+                this.updateCalendarEvent(calendar, item).thenApply(function(res) {
                     that.removeSpinner();
                 }).exceptionally(function(throwable) {
-                    that.showMessage("Unable to save moved event","Please re-create event");
+                    that.showMessage("Unable to save event","Please close calendar and try again");
```

Comment:
what is this handling? what would closing it change?

Reply:
It may fail in the same way that occasionally make folder may fail after creating new account.

Reply:
That is annoying. We should fix that.